plumbing/msg/waiter.go

```diff
@@ -47,9 +47,10 @@ type ChainMessage struct {
 // NewWaiter returns a new Waiter.
 func NewWaiter(chainStore waiterChainReader, messageReader chain.MessageReader, bs bstore.Blockstore, cst *hamt.CborIpldStore) *Waiter {
 	return &Waiter{
-		chainReader: chainStore,
-		cst:         cst,
-		bs:          bs,
+		chainReader:   chainStore,
+		cst:           cst,
+		bs:            bs,
+		messageReader: messageReader,
```

Comment:
We seem to have a lot of names for things that retrieve messages. Do messageReaders and messageProviders have the same interface? Inteface and implementation?

Reply:
Thanks for calling out.  Yeah it's mostly the same thing +/- a method.  I'll choose a standard name and use it throughout.

Reply:
my vote is for MessageProvider if that counts for anything.